core-web/libs/block-editor/src/lib/extensions/dot-placeholder/dot-placeholder-plugin.ts

```diff
@@ -0,0 +1,114 @@
+import { Extension } from '@tiptap/core';
+import { Plugin, PluginKey } from '@tiptap/pm/state';
+import { Decoration, DecorationSet } from '@tiptap/pm/view';
+
+import { NodeTypes } from '../bubble-menu/models';
+
+export interface PlaceholderOptions {
+    emptyEditorClass: string;
+    emptyNodeClass: string;
+    placeholder: string;
+    showOnlyWhenEditable: boolean;
+    showOnlyCurrent: boolean;
+    includeChildren: boolean;
+}
+
+function toTitleCase(str) {
+    return str.replace(/\p{L}+('\p{L}+)?/gu, function (txt) {
+        return txt.charAt(0).toUpperCase() + txt.slice(1);
+    });
+}
+
+export const DotPlaceholder = Extension.create<PlaceholderOptions>({
```

Comment:
We can extend the `Placeholder` plugin so we don't have to create de `addOptions()` and `name` manually.

```suggestion
export const DotPlaceholder = Placeholder.extend<PlaceholderOptions>({
```